src/main/java/com/minecolonies/coremod/colony/buildings/BuildingMiner.java

```diff
@@ -104,39 +105,43 @@
     /**
      * Max depth the miner reaches at level 0.
      */
-    private static final int MAX_DEPTH_LEVEL_0          = 70;
+    private static final int MAX_DEPTH_LEVEL_0 = 70;
 
     /**
      * Max depth the miner reaches at level 1.
      */
-    private static final int MAX_DEPTH_LEVEL_1          = 50;
+    private static final int MAX_DEPTH_LEVEL_1 = 50;
 
     /**
      * Max depth the miner reaches at level 2.
      */
-    private static final int MAX_DEPTH_LEVEL_2          = 30;
+    private static final int MAX_DEPTH_LEVEL_2 = 30;
 
     /**
      * Max depth the miner reaches at level 3.
      */
-    private static final int MAX_DEPTH_LEVEL_3          = 5;
-
+    private static final int MAX_DEPTH_LEVEL_3 = 5;
+    /**
+     * Stores the levels of the miners mine. This could be a map with (depth,level).
+     */
+    @NotNull
+    private final List<Level> levels      = new ArrayList<>();
     /**
      * True if shaft is at bottom limit.
      */
-    public               boolean     clearedShaft       = false;
+    public  boolean clearedShaft       = false;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make clearedShaft a static final constant or non-public and provide accessors if needed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)